mmdet/models/utils/transformer.py

```diff
@@ -0,0 +1,734 @@
+        return_intermediate_dec (bool): Whether to return the intermediate
+            output from each TransformerDecoderLayer or only the last
+            TransformerDecoderLayer. Default False. If False, the returned
+            `hs` has shape [num_decoder_layers, bs, num_query, embed_dims].
+            If True, the returned `hs` will have shape [1, bs, num_query,
+            embed_dims].
+    """
+
+    def __init__(self,
+                 embed_dims=512,
+                 num_heads=8,
+                 num_encoder_layers=6,
+                 num_decoder_layers=6,
+                 feedforward_channels=2048,
+                 dropout=0.0,
+                 act_cfg=dict(type='ReLU', inplace=True),
+                 norm_cfg=dict(type='LN'),
+                 num_fcs=2,
+                 pre_norm=False,
+                 return_intermediate_dec=False):
+        super(Transformer, self).__init__()
+        self.embed_dims = embed_dims
+        self.num_heads = num_heads
+        self.num_encoder_layers = num_encoder_layers
+        self.num_decoder_layers = num_decoder_layers
+        self.feedforward_channels = feedforward_channels
+        self.dropout = dropout
+        self.act_cfg = act_cfg
+        self.norm_cfg = norm_cfg
+        self.num_fcs = num_fcs
+        self.pre_norm = pre_norm
+        self.return_intermediate_dec = return_intermediate_dec
+        if self.pre_norm:
+            encoder_order = ('norm', 'selfattn', 'norm', 'ffn')
+            decoder_order = ('norm', 'selfattn', 'norm', 'multiheadattn',
+                             'norm', 'ffn')
+        else:
+            encoder_order = ('selfattn', 'norm', 'ffn', 'norm')
+            decoder_order = ('selfattn', 'norm', 'multiheadattn', 'norm',
+                             'ffn', 'norm')
+        self.encoder = TransformerEncoder(num_encoder_layers, embed_dims,
+                                          num_heads, feedforward_channels,
+                                          dropout, encoder_order, act_cfg,
+                                          norm_cfg, num_fcs)
+        self.decoder = TransformerDecoder(num_decoder_layers, embed_dims,
+                                          num_heads, feedforward_channels,
+                                          dropout, decoder_order, act_cfg,
+                                          norm_cfg, num_fcs,
+                                          return_intermediate_dec)
+        # follow the official DETR to init parameters
+        self.init_weights()
+
+    def init_weights(self, distribution='uniform'):
+        for m in self.modules():
+            if hasattr(m, 'weight') and m.weight.dim() > 1:
+                xavier_init(m, distribution=distribution)
+
+    def forward(self, x, mask, query_embed, pos_embed):
+        """Forward function for `Transformer`.
+
+        Args:
+            x (Tensor): Input query with shape [bs,c,h,w] where
+                c=embed_dims.
```

Comment:
`c=embed_dims` -> `c = embed_dims` These codes should also follow the python code format.